searchcore/src/vespa/searchcore/proton/metrics/documentdb_tagged_metrics.cpp

```diff
@@ -213,14 +213,15 @@ DocumentDBTaggedMetrics::MatchingMetrics::RankProfileMetrics::update(const Match
     queryLatency.addValueBatch(stats.queryLatencyAvg(), stats.queryLatencyCount(),
                                stats.queryLatencyMin(), stats.queryLatencyMax());
     if (stats.getNumPartitions() > 0) {
-        if (stats.getNumPartitions() <= partitions.size()) {
-            for (size_t i = 0; i < stats.getNumPartitions(); ++i) {
-                partitions[i]->update(stats.getPartition(i));
-            }
-        } else {
-            vespalib::string msg(vespalib::make_string("Num partitions used '%ld' is larger than number of partitions '%ld' configured.",
-                                             stats.getNumPartitions(), partitions.size()));
-            throw vespalib::IllegalStateException(msg, VESPA_STRLOC);
+        for (size_t i = partition.size(); i < stats.getNumPartitions(); ++i) {
+            // This loop is to handle live reconfigs that changes how many partitions(number of threads) might be used per query.
+            vespalib::string partitionId(vespalib::make_string("docid_part%02ld", i));
+            partitions.push_back(std::make_unique<DocIdPartition>(partitionId, this));
```

Comment:
Is this function always called in the context of the `MetricManager` lock? I'm not confident post-construction changes to a `MetricSet` are thread safe otherwise (the code is complicated, so hard to tell for sure without some deeper digging).

Reply:
I will make another pass and check.